src/bosh-director/lib/bosh/director/config_server/variables_interpolator.rb

```diff
@@ -24,7 +25,13 @@ def interpolate_template_spec_properties(template_spec_properties, deployment_na
 
       template_spec_properties.each do |job_name, job_properties|
         begin
-          interpolated_hash = @config_server_client.interpolate_with_versioning(job_properties, variable_set)
+          key = job_name + "-" + instance_name
+          if @cache_by_job_name.has_key?(key)
+            interpolated_hash = @cache_by_job_name[key]
+          else
+            interpolated_hash = @config_server_client.interpolate_with_versioning(job_properties, variable_set)
+            @cache_by_job_name[key] = interpolated_hash
```

Comment:
We don't need to care about synchronisation here because the worst what can happen is to call the config server multiple times. Is this assumption correct?

Reply:
Yup on first call, we get properties from config server and cache just the properties block for each job in an instance group which shouldnt change